presto-main/src/main/java/com/facebook/presto/cost/CoefficientBasedCostCalculator.java

```diff
@@ -74,125 +70,104 @@ public CoefficientBasedCostCalculator(Metadata metadata)
     }
 
     @Override
-    public Map<PlanNodeId, PlanNodeCost> calculateCostForPlan(Session session, Map<Symbol, Type> types, PlanNode planNode)
+    public PlanNodeCost calculateCost(
+            PlanNode planNode,
+            Lookup lookup,
+            Session session,
+            Map<Symbol, Type> types)
     {
-        Visitor visitor = new Visitor(session, types);
-        planNode.accept(visitor, null);
-        return ImmutableMap.copyOf(visitor.getCosts());
+        Visitor visitor = new Visitor(lookup, session, types);
+        return planNode.accept(visitor, null);
     }
 
     private class Visitor
             extends PlanVisitor<PlanNodeCost, Void>
     {
+        private final Lookup lookup;
         private final Session session;
-        private final Map<PlanNodeId, PlanNodeCost> costs;
         private final Map<Symbol, Type> types;
 
-        public Visitor(Session session, Map<Symbol, Type> types)
+        public Visitor(Lookup lookup, Session session, Map<Symbol, Type> types)
         {
-            this.costs = new HashMap<>();
+            this.lookup = lookup;
             this.session = session;
             this.types = ImmutableMap.copyOf(types);
         }
 
-        public Map<PlanNodeId, PlanNodeCost> getCosts()
+        private PlanNodeCost lookupStats(PlanNode sourceNode)
```

Comment:
`s/lookupStats/lookupCost` (unless you squash this commit (or reorder) with "Rename PlanNodeCost to PlanNodeStatsEstimate")